src/repr/src/row/encoding.rs

```diff
@@ -72,73 +44,335 @@ impl Codec for Row {
+                is_tz: false,
+                unknown_fields: Default::default(),
+                cached_size: Default::default(),
+            }),
+            Datum::TimestampTz(x) => {
+                let date = x.date().naive_utc();
+                Datum_type::timestamp(ProtoTimestamp {
+                    year: date.year(),
+                    ordinal: date.ordinal(),
+                    secs: x.time().num_seconds_from_midnight(),
+                    nanos: x.time().nanosecond(),
+                    is_tz: true,
+                    unknown_fields: Default::default(),
+                    cached_size: Default::default(),
+                })
+            }
+            Datum::Interval(x) => {
+                let duration = x.duration.to_le_bytes();
+                let (mut duration_lo, mut duration_hi) = ([0u8; 8], [0u8; 8]);
+                duration_lo.copy_from_slice(&duration[..8]);
+                duration_hi.copy_from_slice(&duration[8..]);
+                Datum_type::interval(ProtoInterval {
+                    months: x.months,
+                    duration_lo: i64::from_le_bytes(duration_lo),
+                    duration_hi: i64::from_le_bytes(duration_hi),
+                    unknown_fields: Default::default(),
+                    cached_size: Default::default(),
+                })
+            }
+            Datum::Bytes(x) => Datum_type::bytes(x.to_vec()),
+            Datum::String(x) => Datum_type::string(x.to_owned()),
+            Datum::Array(x) => Datum_type::array(ProtoArray {
+                elements: MessageField::some(ProtoRow {
+                    datums: x.elements().iter().map(|x| x.into()).collect(),
+                    unknown_fields: Default::default(),
+                    cached_size: Default::default(),
+                }),
+                dims: x
+                    .dims()
+                    .into_iter()
+                    .map(|x| ProtoArrayDimension {
+                        lower_bound: u64::cast_from(x.lower_bound),
+                        length: u64::cast_from(x.length),
+                        unknown_fields: Default::default(),
+                        cached_size: Default::default(),
+                    })
+                    .collect(),
+                unknown_fields: Default::default(),
+                cached_size: Default::default(),
+            }),
+            Datum::List(x) => Datum_type::list(ProtoRow {
+                datums: x.iter().map(|x| x.into()).collect(),
+                unknown_fields: Default::default(),
+                cached_size: Default::default(),
+            }),
+            Datum::Map(x) => Datum_type::dict(ProtoDict {
+                elements: x.iter().map(|(k, v)| (k.to_owned(), v.into())).collect(),
+                unknown_fields: Default::default(),
+                cached_size: Default::default(),
+            }),
+            Datum::Numeric(x) => {
+                // WIP should we do the same reduce that pack_numeric does?
```

Comment:
The reduce operation is only necessary to preserve row equality within Differential, which relies on blithely assessing the rows' bytes. Because this is only storage, it shouldn't matter, and if the data is coming directly from Differential, it should already be reduced.

Reply:
Oops github was hiding this comment from me the first time though. :(

Okay, so it's not required for correctness, is it a good thing to do anyway? We're still writing down these bytes, so if there's something easy to shorten them, it seems like it might be a good idea, but I def defer to you on it.